pkg/client/node.go

```diff
@@ -183,22 +183,34 @@ func NodeAddToCluster(ctx context.Context, runtime runtimes.Runtime, node *k3d.N
 
 	l.Log().Tracef("Resulting node %+v", node)
 
+	var registrationNode *k3d.Node
+	// Use LB if available as registration url for nodes otherwise
+	// fallback to server node
+	for _, existingNode := range cluster.Nodes {
+		if existingNode.Role == k3d.LoadBalancerRole {
+			registrationNode = existingNode
+			break
+		} else if existingNode.Role == k3d.ServerRole {
+			registrationNode = existingNode
+		}
+	}
```

Comment:
Assuming that the list of nodes is somewhat ordered, this will always avoid using server-0 if there's more than one server in a setup with loadbalancer disabled, right?

Reply:
@iwilltry42 Actually the list of nodes is not ordered but rather in the descending order of time added. Here is a order of output that I got from my debugger for my setup:
```
[]*github.com/k3d-io/k3d/v5/pkg/types.Node len: 3, cap: 4, [
        *{
                Name: "k3d-new-0",
                Role: "server",
                ...                                                                                           Memory: "",
                State: (*"github.com/k3d-io/k3d/v5/pkg/types.NodeState")(0xc00042d4a8),
                IP: (*"github.com/k3d-io/k3d/v5/pkg/types.NodeIP")(0xc00042d4d0),                             HookActions: []github.com/k3d-io/k3d/v5/pkg/types.NodeHook len: 0, cap: 0, nil,},
        *{
                Name: "k3d-test-server-1",
                Role: "server",
                ...                                                                                           Memory: "",
                State: (*"github.com/k3d-io/k3d/v5/pkg/types.NodeState")(0xc00042dcc8),
                IP: (*"github.com/k3d-io/k3d/v5/pkg/types.NodeIP")(0xc00042dcf0),                             HookActions: []github.com/k3d-io/k3d/v5/pkg/types.NodeHook len: 0, cap: 0, nil,},
        *{
                Name: "k3d-test-server-0",
                Role: "server",
                ...
                Memory: "",
                State: (*"github.com/k3d-io/k3d/v5/pkg/types.NodeState")(0xc000503988),
                IP: (*"github.com/k3d-io/k3d/v5/pkg/types.NodeIP")(0xc0005039b0),                             HookActions: []github.com/k3d-io/k3d/v5/pkg/types.NodeHook len: 0, cap: 0, nil,},
]
```

Reply:
While debugging the suggested change I did encounter another issue where adding a node to the cluster with no-lb results in segmentation fault. I will create a separate issue for that.